flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ModifyOperation.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.api.QueryConfig;
+import org.apache.flink.table.api.TableSchema;
+
+import java.util.List;
+
+/**
+ * A {@link TableOperation} that describes the DML queries such as e.g. INSERT or conversion to a
```

Comment:
Did we support UPDATE/MERGE/DELETE semantics？ and how?

Reply:
So far we did not support neither of those operations.

In the future, once a `Planner` supports any of them it should return a `DmlTableOperation` that it will understand.